micromanager_gui/main_window.py

```diff
@@ -383,103 +380,6 @@ def _on_config_set(self, groupName: str, configName: str):
-                "Select Objective Device:",
-                self,
-            )
-            obj.val_changed.connect(self._set_objectives)
-            obj.show()
-
-    def _set_objectives(self, obj_device: str):
-
-        obj_dev, obj_cfg, presets = self._get_objective_device(obj_device)
-
-        if obj_dev and obj_cfg and presets:
-            current_obj = self._mmc.getCurrentConfig(obj_cfg)
-        else:
-            current_obj = self._mmc.getState(obj_dev)
-            presets = self._mmc.getStateLabels(obj_dev)
-        self._add_objective_to_gui(current_obj, presets)
-
-    def _get_objective_device(self, obj_device: str):
-        # check if there is a configuration group for the objectives
-        for cfg_groups in self._mmc.getAvailableConfigGroups():
-            # e.g. ('Camera', 'Channel', 'Objectives')
-
-            presets = self._mmc.getAvailableConfigs(cfg_groups)
-
-            if not presets:
-                continue
-
-            # first group option e.g. TINosePiece: State=1
-            cfg_data = self._mmc.getConfigData(cfg_groups, presets[0])
-
-            device = cfg_data.getSetting(0).getDeviceLabel()
-            # e.g. TINosePiece
-
-            if device == obj_device:
-                _core.STATE.objective_device = device
-                _core.STATE.objectives_cfg = cfg_groups
-                return _core.STATE.objective_device, _core.STATE.objectives_cfg, presets
-
-        _core.STATE.objective_device = obj_device
-        return _core.STATE.objective_device, None, None
-
-    def _add_objective_to_gui(self, current_obj, presets):
-        with blockSignals(self.obj_wdg.objective_comboBox):
-            self.obj_wdg.objective_comboBox.clear()
-            self.obj_wdg.objective_comboBox.addItems(presets)
-            if isinstance(current_obj, int):
-                self.obj_wdg.objective_comboBox.setCurrentIndex(current_obj)
-            else:
-                self.obj_wdg.objective_comboBox.setCurrentText(current_obj)
-            self.cam_wdg._update_pixel_size()
-
-    def change_objective(self):
-        if self.obj_wdg.objective_comboBox.count() <= 0:
-            return
-
-        if not _core.STATE.objective_device:
-            return
-
-        zdev = self._mmc.getFocusDevice()
-
-        currentZ = self._mmc.getZPosition()
-        self._mmc.setPosition(zdev, 0)
-        self._mmc.waitForDevice(zdev)
-
-        try:
-            self._mmc.setConfig(
-                _core.STATE.objectives_cfg,
-                self.obj_wdg.objective_comboBox.currentText(),
-            )
-        except ValueError:
-            self._mmc.setProperty(
-                _core.STATE.objective_device,
-                "Label",
-                self.obj_wdg.objective_comboBox.currentText(),
-            )
-
-        self._mmc.waitForDevice(_core.STATE.objective_device)
-        self._mmc.setPosition(zdev, currentZ)
-        self._mmc.waitForDevice(zdev)
-        self.cam_wdg._update_pixel_size()
```

Comment:
Here too: note that the new widget doesn't yet update pixel size when changing the objective... That needs some further consideration